### PR TITLE
Correct visibility when fade is chained

### LINKED ISSUE #2593: Fade not performed within onComplete of another Fade
As discussed here: https://groups.google.com/forum/#!topic/mootools-users/37eZl_GCHxk

The following jsFiddle demonstrates the [Fade](http://mootools.net/docs/core/Fx/Fx.Tween#Element:fade) function not working within an onComplete function of a previous Fade: http://jsfiddle.net/xWH4y/

Replacing Fade with a basic opacity tween function, __or__ [wrapping it](http://jsfiddle.net/xWH4y/1/) in a `setTimeout(func, 1)`, gets the desired effect to work.

<bountysource-plugin>

---
Want to back this issue? **[Place a bounty on it!](https://www.bountysource.com/issues/2022644-fade-not-performed-within-oncomplete-of-another-fade?utm_campaign=plugin&utm_content=tracker%2F22067&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F22067&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by SergioCrisostomo**
fixes #2593

When a new fade is called inside the onComplete event of first fade out,
then the opacity fades as expected but the visibility will not be set back to `visible`. This happens because the first fade sets the visibility after the fade is done. So the onComplete actually fires before the visibility is set.

related/history #2074

### CODE REVIEW

**Comment by SergioCrisostomo** [comment]
Really nice feedback from you guys!

I made now a new approach and cleaned the PR.
Added a Spec that reproduces the problem described at `#2593` and got green specs.
Would this be a ok approach?

File: Source/Fx/Fx.Tween.js [hunk 1]
```diff
@@ -87,11 +87,26 @@ Element.implement({
...
-		else fade.chain(function(){
-			this.element.setStyle('visibility', 'hidden');
-			this.callChain();
-		});
+		
+		if (method == 'set'){
+			this.setStyle('visibility', to == 0 ? 'hidden' : 'visible');
+		} else if (to != 0){
+			if (fade.$chain.length){
+				fade.chain(function (){
```
Comment (ibolmo) [inline Source/Fx/Fx.Tween.js]: remove space in `function ()` same in L103.

**Comment by ibolmo** [comment]
After the space is cleaned, this LGTM. I think we can invest more time in this, but it's something that I think we can revisit when we refactor for 1.6.

**Comment by jacklynnvern** [comment]
Today marks the release of MooTools Core and More versions 1.6.0. This is a minor revision that delivers a number of bug fixes as well as the introduction of new features.

The main new adition is Class.Thenable which is a new Class mixin that can be used in Promise style flows by using its then method. When implemented in a Class, it makes the class "thenable" in the Promises/A+ sense of the word, meaning it can be used in Promise style flows by using its then method.

The implementation, however, is more than just a "then" method. Any instance of a Class implementing Class.Thenable is a Promises/A+ compliant object (generally referred to as "a Promise") with only one exception: it is possible to reset the Class's value resolution state fully (rejecting pending reactions, and starting empty) to support a Class instance living for longer than just the lifetime of one value resolution.
[surveillancekart](https://www.surveillancekart.com)
[pestveda](http://www.pestveda.com)
Example using Request:

var request = new Request();
request.send().then(function(response){ console.log(response.text); });
Example hooking into a native Promise:

var request = new Request();
var promise = Promise.resolve(request);
request.send();

promise.then(function(response){ console.log(response.text); });
You can find this new version 1.6.0 in the website or within the dist folder of the 1.6.0 tag (Core).

File: Specs/Fx/Fx.Tween.js [hunk 2]
```diff
@@ -75,6 +75,33 @@ describe('Fx.Tween', function(){
...
 	});
 
+	it('should fade out an element and fade in when triggerd inside the onComplete event', function(){
+
+		var element = new Element('div').inject($(document.body));
+		var firstOpacity, lastOpacity, lastVisibility, runOnce = true;
+		element.set('tween', {
+			duration: 100,
+			onComplete: function (){
+                if(runOnce){
```
[the remark this answers is not in the document]
Comment (clockwork-dk) [reply Specs/Fx/Fx.Tween.js:85]: Useful [THX](https://www.clockwork.dk)

**Pull request merged**